databricks/koalas/frame.py

```diff
@@ -1545,6 +1546,40 @@ def to_parquet(self, path: str, mode: str = 'error',
         self._sdf.write.parquet(path=path, mode=mode, partitionBy=partition_cols,
                                 compression=compression)
 
+    def to_spark_io(self, path: Optional[str] = None, format: Optional[str] = None,
+                    mode: str = 'error', partition_cols: Union[str, List[str], None] = None,
+                    **options):
+        """Write the DataFrame out to a Spark data source.
+
+        Parameters
+        ----------
+        path : string, optional
+            Path to the data source.
+        format : string, optional
+            Name of the data source in Spark.
```

Comment:
`Name` sounds more like file name if you asked me. Why not instead go for *"Specifies the input data source format."* like the Spark docs describe it.

Also, like you did for `mode`, it would be great to provide a list of supported formats, namely: CSV, JDBC, JSON, ORC, and Parquet.

Reply:
Thanks. Good point. Will do the changes.